pkg/scheduler/reconciliation/persistent.go

```diff
@@ -618,3 +638,23 @@ func (r *PersistentReconciliationRepository) GetAllComponents() ([]string, error
 
 	return components, nil
 }
+
+func splitStringSlice(slice []string, blockSize int) [][]string {
+	sliceLength := len(slice)
+	if sliceLength == 0 {
+		return nil
+	}
+	subSlicesCount := (sliceLength + blockSize - 1) / blockSize
```

Comment:
can we use `math.Ceil` here to avoid the possibility of overflow? since we add two integer

Reply:
Added it to prevent any edge cases including integer overflow issue.

Reply:
in this case, we can rewrite this to `subSlicesCount := (sliceLength - 1) / blockSize + 1` to prevent overflow.

Reply:
That could be another approach sure, added it.